dwave/cloud/coders.py

```diff
@@ -237,4 +242,59 @@ def decode_qp_numpy(msg, return_matrix=True):
         result['active_variables'] = result['active_variables'].tolist()
         result['solutions'] = result['solutions'].tolist()
 
+    # include problem type
+    if 'type' in msg:
+        result['problem_type'] = msg['type']
+
+    return result
+
+
+def encode_problem_as_bq(problem, compress=False):
+    """Encode the binary quadratic problem for submission the `bq` data format.
```

Comment:
Accidentally noticed what I think is a missing "in". Also, we usually use "binary quadratic model" or problem but not "binary quadratic problem", typically the BQM represents a reformulation of he user's problem, so I think it's helpful to keep that distinction.

Reply:
Fixed the "in" typo, thanks.

Re: bqp, I see the term is used elsewhere, and it was heavily used in the qbsolv/microclient era. Even the encoding format on SAPI is called `qp` (quadratic problem).

If we want to use the new terminology in CC, we should consistently change it. And that's a documentation refactor I mentioned yesterday.